cmd/performance-profile-creator/cmd/root.go

```diff
@@ -127,25 +128,48 @@ func getDataFromFlags(cmd *cobra.Command) (profileCreatorArgs, error) {
 }
 
 func getProfileData(args profileCreatorArgs) (*ProfileData, error) {
-	mcp, err := profilecreator.GetMCP(args.mustGatherDirPath, args.mcpName)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get MachineConfigPool with mcp-name %s: %v", args.mcpName, err)
+	info, err := os.Stat(args.mustGatherDirPath)
+	if os.IsNotExist(err) {
+		return nil, fmt.Errorf("the must-gather path '%s' is not valid", args.mustGatherDirPath)
 	}
-
-	nodes, err := profilecreator.GetNodeList(args.mustGatherDirPath)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get Nodes with mcp-name %s: %v", args.mcpName, err)
+		return nil, fmt.Errorf("can't access the must-gather path: %v", err)
+	}
+	if !info.IsDir() {
+		return nil, fmt.Errorf("the must-gather path '%s' is not a directory", args.mustGatherDirPath)
 	}
 
 	mcps, err := profilecreator.GetMCPList(args.mustGatherDirPath)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get the MCP list under %s: %v", args.mustGatherDirPath, err)
 	}
 
+	var mcp *machineconfigv1.MachineConfigPool
+	mcpNames := make([]string, 0)
+	for _, m := range mcps {
+		mcpNames = append(mcpNames, m.Name)
+		if m.Name == args.mcpName {
```

Comment:
Suggestion: once we find a match, we can break from the loop

Reply:
We can't, because we are accumulating all the mcpNames in the cluster in order to help the user:
Couldn't find the the mcp X, try one of [mcpNames]

Reply:
Okay, got it! Thanks for the clarification, that makes sense!

Reply:
Actually, if we did find the VM we don't need the mcpNames anyway since we have the right MCP. However there may be other reasons we can't use the mcp: no nodes defined and so on. So we better keep it anyway. Also we may re-use the mcp names for other info  logs.

Reply:
Ack.